public/main.js

```diff
@@ -1,14 +1,35 @@
-function fetchGET(url, cb) {
-    var xhr = new XMLHttpRequest();
-    xhr.onreadystatechange = function() {
-        if (xhr.readySate === 4 && xhr.status === 200) {
-            cb(null, JSON.parse(xhr.responseText));
-        } else if (xhr.readyState === 4 && xhr.status !== 200) {
-            cb(xhr.status);
-        } else {
-            cb(null, xhr.responseText);
-        }
-    };
-    xhr.open('GET', url, true);
-    xhr.send();
+function fetchPOST(url, data, callback) {
+  var xhr = new XMLHttpRequest();
+  xhr.onreadystatechange = function() {
+    if (xhr.readyState == 4 && xhr.status !== 200) {
+      callback(xhr.responseText);
+    } else if (xhr.readyState == 4 && xhr.status === 200) {
+      callback(null, xhr.responseText);
+    }
+  };
+  xhr.open('POST', url);
+  xhr.setRequestHeader('Content-Type', 'application/json');
+  xhr.send(JSON.stringify(data));
 }
+
+
+var buttons = document.querySelectorAll('.article__btn');
+
+buttons.forEach(function(btn) {
+  btn.addEventListener('click', function(e) {
+    e.preventDefault();
+    var query = [btn.parentElement[0].value];
+    console.log(query);
+    var data = fetchPOST('/visualize', query, function(err, data) {
+      if (err) {
+        console.log(err);
+      }
+      else {
+        //This is the time to process the data using D3.js
+        console.log(data);
```

Comment:
Get rid of the console.log

Reply:
We need the console log for now to help the people working on the data rendering.

Reply:
They can write the console.log in themselves 😃 . The comment here is sufficient as a TODO note

Reply:
Ok cool will correct